stacker/blueprints/base.py

```diff
@@ -152,6 +173,95 @@ def setup_parameters(self):
             p = build_parameter(param, attrs)
             t.add_parameter(p)
 
+    def defined_parameters(self):
+        """Return a dictionary of parameters defined by the blueprint.
+
+        By default, this will just return the values from
+        `BLUEPRINT_PARAMETERS`, but this makes it easy for subclasses to add
+        parameters.
+
+        Returns:
+            dict: parameters defined by the blueprint
+
+        """
+        return getattr(self, "BLUEPRINT_PARAMETERS", {})
+
+    def get_parameters(self):
+        """Return a dictionary of parameters available to the template.
+
+        These parameters will have been defined within `BLUEPRINT_PARAMETERS`
+        or `self.defined_parameters`. Any parameter value that depends on stack
+        output will have been resolved.
+
+        Returns:
+            dict: parameters available to the template
+
+        """
+        if self.resolved_parameters is None:
+            raise UnresolvedBlueprintParameters(self)
+        return self.resolved_parameters
+
+    def resolve_parameters(self, parameter_values, provider, context):
+        """Resolve the values of the blueprint parameters.
+
+        resolve_parameters will resolve the values of the
+        `BLUEPRINT_PARAMETERS` with values from the command lie, the env file,
+        the config, and any references to other stack output.
+
+        Args:
+            parameter_values (dict): dictionary of parameter key to parameter
+                value
+            provider (stacker.Provider): provider to use to resolve references
```

Comment:
Can you use this syntax for classes - it makes the API docs easier to use: https://github.com/remind101/stacker/blob/700ec034f2a9ef5711b8c21b5705eb9d5993b091/stacker/actions/build.py#L24